src/reanimated2/platform-specific/checkPluginVersion.ts

```diff
@@ -0,0 +1,30 @@
+import { jsVersion } from './jsVersion';
+
+export function checkPluginVersion(): void {
+  const pluginVersion = global._REANIMATED_VERSION_PLUGIN;
+  if (pluginVersion === undefined) {
+    console.error(
+      `[Reanimated] Couldn't determine the version of babel plugin of Reanimated. Did you forget to add 'react-native-reanimated/plugin' to your 'babel.config.js' file?`
+    );
+    return;
+  }
+  const ok = (() => {
+    if (
+      jsVersion.match(/^\d+\.\d+\.\d+$/) &&
+      pluginVersion.match(/^\d+\.\d+\.\d+$/)
+    ) {
+      // x.y.z, compare only major and minor, skip patch
+      const [jsMajor, jsMinor] = jsVersion.split('.');
+      const [pluginMajor, pluginMinor] = pluginVersion.split('.');
+      return jsMajor === pluginMajor && jsMinor === pluginMinor;
+    } else {
+      // alpha, beta or rc, compare everything
+      return jsVersion === pluginVersion;
+    }
+  })();
```

Comment:
Duplicated code, please move it to some utils file and write some unit tests as well.